### PR TITLE
build: add a workaround for CMake 3.20

### PR DESCRIPTION
**Pull request by compnerd**
CMake 3.20 had a bug on the release where the user supplied paths would
be processed without conversion.  This adds a workaround by using
`cygpath` to convert the path to a mixed model style which should avoid
the problem.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
CC: @drodriguez

**Comment by compnerd** [comment]
@swift-ci please smoke test

File: utils/build-windows.bat [hunk 1]
```diff
@@ -1,376 +1,379 @@
...
-goto :eof
-endlocal
-
-
-:end
+:: build-windows.bat
+::
+:: This source file is part of the Swift.org open source project
+::
+:: Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
```
Comment (shahmishal) [inline utils/build-windows.bat:5]: Can you update this to 2021?

Comment (shahmishal) [reply utils/build-windows.bat:5]:
```suggestion
:: Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
```

**Comment by drodriguez** [comment]
The change looks fine.

Just a warning: I think the newlines of the file had changed. If I remember correctly, `*.bat` files needed to be CRLF for some reason. It seems to be working fine in CI, but if something strange starts to happen, maybe switching back to CRLF might fix it.

**Comment by compnerd** [comment]
@drodriguez thanks for that careful review!  I've put up #38992 to ensure that this problem doesn't happen.

**Pull request merged**